### PR TITLE
Ndev 3386 caching build config

### PR DESCRIPTION
**Pull request by rnovikov**

### CODE REVIEW

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/12161530155/artifacts/2273936653)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| simulateTransaction               |  438964 |      0     |      0.132 |    0.000730329 |         0.001 |
| getAccountInfo                    |   87810 |      0     |      0.137 |    0.000878009 |         0.001 |
| getSlot                           |   85571 |      0     |      0.163 |    0.000620853 |         0.001 |
| getBlocks                         |   41189 |      0     |      0.101 |    0.000588725 |         0.001 |
| getBlock                          |   20807 |      0     |      0.049 |    0.00150594  |         0.001 |
| getMultipleAccounts               |   16914 |      0     |      0.146 |    0.00173507  |         0.001 |
| getLatestBlockhash                |   12805 |      0     |      0.126 |    0.00176868  |         0.001 |
| getTransaction                    |    9917 |      0     |      0.165 |    0.00965957  |         0.003 |
| getBlockTime                      |    6225 |      0     |      0.048 |    0.000710361 |         0     |
| sendTransaction                   |    5147 |      0     |      0.146 |    0.00943734  |         0.005 |
| getBalance                        |    1770 |      0     |      0.022 |    0.00104068  |         0.001 |
| getSignatureStatuses              |    1160 |      0     |      0.022 |    0.000965517 |         0.001 |
| getVersion                        |     606 |      0     |      0.022 |    0.000765677 |         0.001 |
| isBlockhashValid                  |     562 |      0     |      0.005 |    0.000562278 |         0     |
| getSignaturesForAddress           |      62 |      0     |      0.004 |    0.00209677  |         0.002 |
| requestAirdrop                    |      32 |      0     |      0.002 |    0.00090625  |         0.001 |
| getTokenAccountBalance            |       5 |      0     |      0.001 |    0.0004      |         0     |
| getHealth                         |       3 |      0     |      0.001 |    0.000666667 |         0.001 |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0     |      0     |    0           |         0     |
| getClusterNodes                   |       1 |      0     |      0     |    0           |         0     |

</details>

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/12162996807/artifacts/2274431787)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |  100271 |      0     |      0.125 |    0.00063483  |         0.001 |
| getAccountInfo                    |   62611 |      0     |      0.136 |    0.000815735 |         0.001 |
| getBlocks                         |   48679 |      0     |      0.059 |    0.000601245 |         0.001 |
| getBlock                          |   24576 |      0     |      0.029 |    0.00147343  |         0.001 |
| getMultipleAccounts               |   18662 |      0     |      0.102 |    0.0012145   |         0.001 |
| getLatestBlockhash                |   14279 |      0     |      0.12  |    0.00103894  |         0.001 |
| getTransaction                    |    9835 |      0     |      0.131 |    0.00760203  |         0.003 |
| getBlockTime                      |    6240 |      0     |      0.056 |    0.000516987 |         0     |
| sendTransaction                   |    5102 |      0     |      0.138 |    0.00653332  |         0.004 |
| getBalance                        |    1779 |      0     |      0.111 |    0.00108657  |         0.001 |
| getSignatureStatuses              |    1121 |      0     |      0.135 |    0.00100892  |         0.001 |
| getVersion                        |     601 |      0     |      0.054 |    0.00111481  |         0     |
| isBlockhashValid                  |     542 |      0     |      0.007 |    0.000457565 |         0     |
| simulateTransaction               |     234 |      0     |      0.003 |    0.0011453   |         0.001 |
| getSignaturesForAddress           |      59 |      0     |      0.004 |    0.00176271  |         0.002 |
| requestAirdrop                    |      32 |      0     |      0.002 |    0.001       |         0.001 |
| getTokenAccountBalance            |      12 |      0     |      0.061 |    0.00566667  |         0.001 |
| getHealth                         |       3 |      0.001 |      0.001 |    0.001       |         0.001 |
| getFirstAvailableBlock            |       2 |      0.001 |      0.001 |    0.001       |         0.001 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |         0.001 |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |         0.001 |

</details>

File: evm_loader/lib/src/rpc/validator_client.rs [hunk 1]
```diff
@@ -114,6 +114,9 @@ impl Deref for CloneRpcClient {
 
 #[async_trait(?Send)]
 impl Rpc for CloneRpcClient {
+    async fn get_last_deployed_slot(&self, _program_id: &Pubkey) -> ClientResult<Option<u64>> {
```
Comment (s-medvedev) [inline evm_loader/lib/src/rpc/validator_client.rs]: Please, add a comment as to why this function returns this value.

File: evm_loader/lib/src/types/programs_cache.rs [hunk 2]
```diff
@@ -18,17 +19,52 @@ use std::sync::RwLock;
...
+    pub fn new() -> Self {
+        Self {
+            table: RwLock::new(HashMap::new()),
+        }
+    }
+
+    fn get(&self, key: &KeyAccountCache) -> Option<Value> {
+        self.table
+            .read()
+            .expect("acc_hash_get_instance poisoned")
```
Comment (s-medvedev) [inline evm_loader/lib/src/types/programs_cache.rs]: Strange error message

File: evm_loader/lib/src/types/programs_cache.rs [hunk 3]
```diff
@@ -18,17 +19,52 @@ use std::sync::RwLock;
...
+            .get(key)
+            .cloned()
+    }
+    fn add(&self, key: KeyAccountCache, value: Value) {
+        self.table
+            .write()
+            .expect("PANIC, no space ")
```
Comment (s-medvedev) [inline evm_loader/lib/src/types/programs_cache.rs]: Strange error message

File: evm_loader/lib/src/types/programs_cache.rs (hunk 2 above)
Comment (rnovikov) [reply evm_loader/lib/src/types/programs_cache.rs]: Done

File: evm_loader/lib/src/types/programs_cache.rs (hunk 3 above)
Comment (rnovikov) [reply evm_loader/lib/src/types/programs_cache.rs]: Done

File: evm_loader/lib/src/rpc/validator_client.rs (hunk 1 above)
Comment (rnovikov) [reply evm_loader/lib/src/rpc/validator_client.rs]: Please take a look on comment

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/12174141818/artifacts/2277591451)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   61586 |      0     |      0.176 |    0.000885623 |        0.001  |
| getSlot                           |   58618 |      0     |      0.14  |    0.000608414 |        0      |
| getBlocks                         |   27745 |      0     |      0.083 |    0.000556461 |        0.001  |
| getMultipleAccounts               |   15208 |      0     |      0.169 |    0.00149638  |        0.001  |
| getBlock                          |   14056 |      0     |      0.08  |    0.0015397   |        0.001  |
| getLatestBlockhash                |   10144 |      0     |      0.122 |    0.00120179  |        0.001  |
| getTransaction                    |    9931 |      0     |      0.179 |    0.00915688  |        0.003  |
| getBlockTime                      |    6234 |      0     |      0.038 |    0.000522297 |        0      |
| sendTransaction                   |    5151 |      0     |      0.142 |    0.00865055  |        0.004  |
| getBalance                        |    1776 |      0     |      0.078 |    0.00111543  |        0.001  |
| getSignatureStatuses              |    1136 |      0     |      0.046 |    0.000987676 |        0.001  |
| getVersion                        |     608 |      0     |      0.064 |    0.00115954  |        0      |
| isBlockhashValid                  |     550 |      0     |      0.009 |    0.000483636 |        0      |
| simulateTransaction               |     180 |      0     |      0.004 |    0.00177222  |        0.002  |
| getSignaturesForAddress           |      58 |      0     |      0.004 |    0.00194828  |        0.002  |
| requestAirdrop                    |      32 |      0     |      0.001 |    0.00096875  |        0.001  |
| getTokenAccountBalance            |      12 |      0     |      0.016 |    0.00441667  |        0.001  |
| getHealth                         |       3 |      0     |      0.004 |    0.00166667  |        0.001  |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |        0.001  |
| getClusterNodes                   |       1 |      0     |      0     |    0           |        0      |

</details>

File: evm_loader/lib/src/rpc/validator_client.rs (hunk 1 above)
Comment (rnovikov) [reply evm_loader/lib/src/rpc/validator_client.rs]: Discussed on meet

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/12175435432/artifacts/2277918311)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getSlot                           |   63276 |      0     |      0.097 |    0.000583286 |        0      |
| getAccountInfo                    |   61055 |      0     |      0.123 |    0.000716518 |        0      |
| getBlocks                         |   28411 |      0     |      0.071 |    0.000526451 |        0      |
| getMultipleAccounts               |   15268 |      0     |      0.078 |    0.00153203  |        0.001  |
| getBlock                          |   14339 |      0     |      0.022 |    0.00133538  |        0.001  |
| getLatestBlockhash                |   10265 |      0     |      0.337 |    0.00166722  |        0.001  |
| getTransaction                    |    9695 |      0     |      0.264 |    0.00893935  |        0.003  |
| getBlockTime                      |    6195 |      0     |      0.048 |    0.000403067 |        0      |
| sendTransaction                   |    5032 |      0     |      0.098 |    0.00840103  |        0.004  |
| getBalance                        |    1762 |      0     |      0.103 |    0.00101362  |        0      |
| getSignatureStatuses              |    1122 |      0     |      0.075 |    0.000894831 |        0.001  |
| getVersion                        |     595 |      0     |      0.053 |    0.00105546  |        0      |
| isBlockhashValid                  |     543 |      0     |      0.009 |    0.000399632 |        0      |
| simulateTransaction               |     504 |      0     |      0.008 |    0.00394048  |        0.004  |
| getSignaturesForAddress           |      60 |      0     |      0.005 |    0.00208333  |        0.002  |
| requestAirdrop                    |      32 |      0     |      0.003 |    0.000875    |        0.001  |
| getTokenAccountBalance            |       9 |      0     |      0.018 |    0.00244444  |        0      |
| getHealth                         |       3 |      0     |      0.001 |    0.000333333 |        0      |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |    0.0005      |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |    0.001       |        0.001  |
| getClusterNodes                   |       1 |      0.001 |      0.001 |    0.001       |        0.001  |

</details>

**Review by s-medvedev — APPROVED** [review]

**Comment by s-medvedev** [comment]
Wait for @anton-lisanin review

**Review by s-medvedev — CHANGES_REQUESTED** [review]
Looks good for me. But Wait for @anton-lisanin review and merge Solana signatures #559

**Comment by neonlabstech** [comment]
<details><summary>Dapps report</summary>

🔗[Cost report](https://github.com/neonlabsorg/neon-proxy.py/actions/runs/12257180412/artifacts/2300046792)

**Comment by neonlabstech** [comment]
<details><summary>Solana Requests Statistics</summary>



|                                   |   count |   min_time |   max_time |   average_time |   median_time |
|:----------------------------------|--------:|-----------:|-----------:|---------------:|--------------:|
| getAccountInfo                    |   64882 |      0     |      0.295 |     0.00285776 |        0.001  |
| getSlot                           |   47203 |      0     |      0.261 |     0.00143953 |        0.001  |
| getBlocks                         |   24486 |      0     |      0.078 |     0.00125676 |        0.001  |
| getMultipleAccounts               |   15712 |      0     |      0.261 |     0.00368648 |        0.002  |
| getBlock                          |   14726 |      0.001 |      0.179 |     0.00439861 |        0.002  |
| getLatestBlockhash                |   10670 |      0     |      0.213 |     0.0028149  |        0.001  |
| getTransaction                    |   10456 |      0.001 |      0.268 |     0.017266   |        0.007  |
| getBlockTime                      |    6298 |      0     |      0.166 |     0.00207177 |        0.001  |
| sendTransaction                   |    5307 |      0.001 |      0.297 |     0.0146731  |        0.008  |
| getBalance                        |    1803 |      0     |      0.235 |     0.00343372 |        0.001  |
| getSignatureStatuses              |    1182 |      0     |      0.149 |     0.00339255 |        0.001  |
| getVersion                        |     631 |      0     |      0.178 |     0.00377971 |        0.001  |
| isBlockhashValid                  |     573 |      0     |      0.03  |     0.00196859 |        0.001  |
| simulateTransaction               |     560 |      0     |      0.047 |     0.00669107 |        0.005  |
| getSignaturesForAddress           |      60 |      0.001 |      0.019 |     0.00361667 |        0.003  |
| requestAirdrop                    |      32 |      0     |      0.009 |     0.0019375  |        0.002  |
| getTokenAccountBalance            |      12 |      0     |      0.01  |     0.00308333 |        0.001  |
| getHealth                         |       5 |      0.001 |      1.029 |     0.2092     |        0.001  |
| getFirstAvailableBlock            |       2 |      0     |      0.001 |     0.0005     |        0.0005 |
| getMinimumBalanceForRentExemption |       1 |      0.001 |      0.001 |     0.001      |        0.001  |
| getClusterNodes                   |       1 |      0.022 |      0.022 |     0.022      |        0.022  |

</details>

**Review by s-medvedev — APPROVED** [review]

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**